drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,403 @@
+	volatile uint8_t *reg_vcmpxcselm = (uint8_t *)config->reg_vcmpxcselm;
+	volatile uint8_t *reg_vcmpscp = (uint8_t *)config->reg_vcmpscp;
+
+	/* Disable voltage comparator specific channel before init */
+	vcmp_enable(dev, 0);
+
+	/*
+	 * ADC channel signal output to voltage comparator,
+	 * so we need to set ADC channel to alternate mode first.
+	 */
+	adc_channel_setup(config->adc, &data->adc_ch_cfg);
+
+	/* Select which ADC channel output voltage into comparator */
+	if (data->adc_ch_cfg.channel_id <= 7) {
+		/* ADC channel 0~7 map to value 0x0~0x7 */
+		*reg_vcmpxctl |= data->adc_ch_cfg.channel_id & VCMP_REG_MASK;
+		*reg_vcmpxcselm &= ~IT8XXX2_VCMP_VCMPXCSELM;
+	} else {
+		/* ADC channel 13~16 map to value 0x8~0xb */
+		*reg_vcmpxctl |= (data->adc_ch_cfg.channel_id - 5) & VCMP_REG_MASK;
+		*reg_vcmpxcselm |= IT8XXX2_VCMP_VCMPXCSELM;
+	}
+
+	/* Set minimum scan period for "all voltage comparator channel" */
+	if (*reg_vcmpscp > config->scan_period) {
+		*reg_vcmpscp = config->scan_period;
+	}
+
+	/* Data must keep device reference for worker handler */
+	data->vcmp = dev;
+
+	/* Init and set worker queue to enable notifications */
+	k_work_init(&data->work, it8xxx2_vcmp_trigger_work_handler);
+	vcmp_work_addr[config->vcmp_ch] = (uint32_t) &data->work;
+
+	/* Set threshold and comparison if set on device tree */
+	if ((config->threshold_mv != IT8XXX2_VCMP_UNDEFINED) &&
+	     (config->comparison != IT8XXX2_VCMP_UNDEFINED)) {
+		enum sensor_attribute attr;
+		struct sensor_value val;
+
+		if (config->comparison == IT8XXX2_VCMP_LESS_OR_EQUAL) {
+			attr = SENSOR_ATTR_LOWER_VOLTAGE_THRESH;
+		} else {
+			attr = SENSOR_ATTR_UPPER_VOLTAGE_THRESH;
+		}
+
+		val.val1 = config->threshold_mv;
+		val.val2 = 0;
+
+		vcmp_ite_it8xxx2_attr_set(dev, SENSOR_CHAN_VOLTAGE, attr, &val);
+	}
+
+	ite_intc_isr_clear(config->irq);
+
+	irq_connect_dynamic(DT_INST_IRQN(0), 0,
+			    (void (*)(const void *))vcmp_it8xxx2_isr,
+			    (const void *)dev, 0);
+
+	irq_enable(config->irq);
+
+	return 0;
+}
+
+static const struct sensor_driver_api vcmp_ite_it8xxx2_api = {
+	.attr_set = vcmp_ite_it8xxx2_attr_set,
+	.attr_get = vcmp_ite_it8xxx2_attr_get,
+	.trigger_set = vcmp_ite_it8xxx2_trigger_set,
+	.channel_get = vcmp_it8xxx2_channel_get,
+};
+
+#define VCMP_IT8XXX2_INIT(inst)						       \
+	static const struct vcmp_it8xxx2_config vcmp_it8xxx2_cfg_##inst = {    \
+		.reg_vcmpxctl = DT_INST_REG_ADDR_BY_IDX(inst, 0),	       \
+		.reg_vcmpxcselm = DT_INST_REG_ADDR_BY_IDX(inst, 1),	       \
+		.reg_vcmpscp = DT_INST_REG_ADDR_BY_IDX(inst, 2),	       \
+		.reg_vcmpxthrdatm = DT_INST_REG_ADDR_BY_IDX(inst, 3),	       \
+		.reg_vcmpxthrdatl = DT_INST_REG_ADDR_BY_IDX(inst, 4),	       \
+		.reg_vcmpsts = DT_INST_REG_ADDR_BY_IDX(inst, 5),	       \
+		.reg_vcmpsts2 = DT_INST_REG_ADDR_BY_IDX(inst, 6),	       \
+		.irq = DT_INST_IRQN(0),					       \
+		.vcmp_ch = DT_INST_PROP(inst, vcmp_ch),			       \
+		.adc = DEVICE_DT_GET(DT_INST_PROP(inst, adc)),		       \
```

Comment:
There are dedicated DT macros for ADC: look at include/devicetree/io-channels.h

Reply:
Done